### PR TITLE
[fix]: [PL-28077]: Added Logs for UnknownHost Exception and handled error for LDAP

### PR DESCRIPTION
**Pull request by adiyaar24**
This PR Handles the errors sent from Delegates to LDAP for unknownHost Exception message .

Test Done
- Created a LDAP Connection with Valid Hostname and port and other credentials . Result was a success.
- Create a LDAP Connection with invalid host ( abc.testy.in) . Result was an error message stating invalid host and port . 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51449)

### CODE REVIEW

**Comment by adiyaar24** [comment]
trigger uts

**Comment by adiyaar24** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/executioncapability/SocketConnectivityCapabilityCheck.java [hunk 1]
```diff
@@ -69,6 +70,10 @@ public static boolean connectableHost(String host, int port) {
       socket.connect(new InetSocketAddress(host, port), 5000); // 5 sec timeout
       log.info("[Delegate Capability] Socket Connection Succeeded for url {} on port {}", host, port);
       return true;
+
+    } catch (UnknownHostException unknownHostException) {
+      log.info("Invalid Host or Port provided {}:{} : Error {}", host, port,
```
Comment (pbarapatre10) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/executioncapability/SocketConnectivityCapabilityCheck.java:75]: This should be `log.warn`. Also, any other code flow where we can get the same UnknownHostException? 
Please get it reviewed from @AnupamIO or @jennyjames21 as well from delegate end, thanks

File: 120-ng-manager/modules/ldap/src/test/java/io/harness/ldap/service/impl/NGLdapServiceImplTest.java [hunk 2]
```diff
@@ -154,6 +155,24 @@ public void testLdapConnectionSuccessfulAndUnsuccessful() throws IOException {
     }
   }
 
+  @Test
+  @Owner(developers = ADITYA)
+  @Category(UnitTests.class)
+  public void testWithInvalidHost() throws IOException {
+    final String accountId = "testAccountId";
+
+    software.wings.beans.sso.LdapSettings ldapSettings = getLdapSettings(accountId);
```
Comment (AnupamIO) [inline 120-ng-manager/modules/ldap/src/test/java/io/harness/ldap/service/impl/NGLdapServiceImplTest.java:164]: import software.wings.beans.sso.LdapSettings

**Review by AnupamIO — APPROVED** [review]

**Comment by adiyaar24** [comment]
trigger ut5

**Review by pbarapatre10 — APPROVED** [review]

**Comment by adiyaar24** [comment]
trigger ut6

**Comment by adiyaar24** [comment]
trigger ut7

**Comment by adiyaar24** [comment]
trigger ut8

**Comment by adiyaar24** [comment]
trigger ut9

**Pull request merged**